Add a package-loading version of runPython

mdboom opened the pull request:
This utilizes a proposed new API in Iodide to allow language evaluators to be asynchronous, which will allow pyodide to do asynchronous things, like loading packages immediately before they are imported.

See https://github.com/iodide-project/iodide/pull/1038 and https://github.com/iodide-project/iodide/pull/1025

rth commented on src/pyodide.py:
```diff
@@ -39,4 +39,22 @@ def eval_code(code, ns):
+        elif isinstance(node, ast.ImportFrom):
+            name = node.module
+            imports.add(name.split('.')[0])
+    return list(imports)
```
Maybe it would be worth adding some test for this? For instance,

**conftest.py**
```py

@pytest.fixture
def noselenium():
    """A fixture to allow running tests that don't require selenium"""
    return 'noselenium'
```

**test/test_package_loading.py**
```py
def test_import_determination(noselenium):
   imports_list = find_imports(
        """
        import numpy as np
        np.arange(10)
        """)
   assert imports_list == ['numpy']
   # etc
```

then create yet another CI test job with,
```sh
pytest -v tests -k noselenium
```
to run tests that don't need selenium.

It's a bit awkward to have this `noselenium` fixture, I'm not sure there is a better way.  I think sooner or later we would need some mechanisms to run pure python tests anyway..

rth replied:
Also maybe add a test to check that this works, something like,
```py
def test_load_packages_multiple(selenium_standalone):
    selenium = selenium_standalone
    selenium.run_async("""
         import pyparsing
         """)
```
where `run_async` is the appropriately defined method? Or is it problematic to do that because it's async?

mdboom commented:
Thanks for pointing out the dichotomy of package names and import names.  That gets me every time.  :)  I've fixed this by generating a mapping (from `test/imports`) from import names to package names that's used to generate the list of packages to load.

I've also added testing of the various accept and failure modes for `runPythonAsync`.

We could test the Python directly, but I'm not sure it's worth adding all that testing complexity when it's tested "under the hood" by the selenium integration tests.

mdboom commented:
I created #218 to remind us to come back around to pure Python testing in the future.
